Revit_Core_Engine/Query/ClosestElement.cs

```diff
@@ -0,0 +1,125 @@
+        [Input("builtInCategory", "Optional, the Revit BuiltInCategory to filter elements in the closest elements search.")]
+        [Input("includeLinkInstance", "Optional: if true, elements from linked documents will also be taken into account.")]
+        [Output("closestElement", "The closest element from the input element.")]
+        public static Element ClosestElement(this Document document, Element originalElement, double searchRadius = Double.MaxValue, BuiltInCategory category = default, bool includeLinks = false)
+        {
+            if (document == null || originalElement == null)
+                return null;
+            
+            List<Document> documentsToSearch = new List<Document>(){document};
+            if (includeLinks)
+            {
+                List<ElementId> linkInstancesIds = Core.Query.ElementIdsOfLinkInstances(document);
+                if (linkInstancesIds.Any())
+                {
+                    foreach (ElementId id in linkInstancesIds)
+                    {
+                        try
+                        {
+                            RevitLinkInstance linkInstance = document.GetElement(id) as RevitLinkInstance;
+                            Document doc = linkInstance.GetLinkDocument();
+                            // Linked files that are not loaded will be null
+                            if(doc != null)
+                                documentsToSearch.Add(doc);
+                        }
+                        catch (Exception e){}
+                    }
+                }
+            }
+
+            // get original element location
+            Location location = originalElement.Location;
+            if (!(location is LocationPoint))
+            {
+                BH.Engine.Reflection.Compute.RecordError(String.Format("Original element to search for closest element is not point XYZ based. Location of type {0} is not implemented in this search.",location.GetType()));
+                return null;
+            }
+            XYZ point = (location as LocationPoint).Point;
+            
+            double smallestDistance = searchRadius;
+            Element result = null;
+            
+            foreach (Document doc in documentsToSearch)
+            {
+                FilteredElementCollector filteredElementCollector = new FilteredElementCollector(doc).WhereElementIsNotElementType();
+                if (category != default)
+                {
+                    filteredElementCollector.OfCategory(category);
+                }
+                
+                IEnumerable<Element> elements = filteredElementCollector.Where(x=>x.HasLocation(true,false));
+
+                //if document is from linked file we need to make sure location is corrected
+                Transform transform = null;
+                if(doc.LinkInstance() != null)
+                    transform = doc.LinkTransform();
```

Comment:
Sorry I have one more comment: to make the code a bit cleaner and avoid the link being host document, you could make the following change:
```suggestion
                if(doc != document)
                    transform = doc.LinkTransform();
```